doc/tutorial.md

```diff
@@ -0,0 +1,212 @@
+Alternatively, we can generate and assign a universally unique identifier (UUID). In Python, this can be achieved by the use of the [UUID module](https://docs.python.org/3/library/uuid.html):
+
+```python
+import uuid
+
+# Generate a random UUID
+entity_uuid = uuid.uuid4()
+
+with open("myEntity.json", "w") as entity_file:
+    entity_file.write('"uuid": {}'.format(entity_uuid))
+```
+Our json file now contains the following:
+
+```json
+"uuid": <uuid>
+```
+where `<uuid>` is a unique identifier for our Entity. 
+
+
+### **Step 2**: Providing a link to the metadata
+Let us assume that we have followed option 1 and provided a URI. The next step consists in adding a **meta** field that will link the Entity to its metadata. In DLite, all Entities are instances of the *EntitySchema*. The *EntitySchema* defines the structure of the Entity. In the **meta** field of our Entity, we need to provide the URI of the EntitySchema. For a schematic overview of the DLite data structures and further detail, see [Concepts section of the DLite User Guide](https://sintef.github.io/dlite/user_guide/concepts.html). We add this information to our json file in the following manner:
+```json
+"uri": "http://www.ontotrans.eu/0.1/solarPanelMeasurement",
+"meta": "http://onto-ns.com/meta/0.3/EntitySchema",
+```
+
+### **Step 3**: Adding a human-understandable description
+Next, we want to include a human-understandable description of what the Entity represents. In our example case, such a **description** field could be
+
+```json
+"uri": "http://www.ontotrans.eu/0.1/solarPanelMeasurement",
+"meta": "http://onto-ns.com/meta/0.3/EntitySchema",
+"description": "Measurement data from one solar panel."
+```
+
+### **Step 4**: Defining the dimensions
+We continue by defining the dimensions for our Entity. Each dimension should have a **name** and a human-understandable **description**. In our example case, we only have one dimension since all the quantities we are interested in are measured with the same frequency (i.e., we have the same number of data points for all the measured quantities). Note that even though we only have one dimension, we need to give it in a list format.
+
+We will give our dimension the generic name "N", and describe it as the number of measurements:
+
+```json
+"uri": "http://www.ontotrans.eu/0.1/solarPanelMeasurement",
+"meta": "http://onto-ns.com/meta/0.3/EntitySchema",
+"description": 
+ "dimensions": [
+      {
+      "name": "N",
+      "description": "Number of measurements."
+      }
+    ]
+```
+
+### **Step 5**: Defining the properties
+Now it is time to define the properties of our Entity. Here is where we can give meaning to our data. As for the dimensions, we add the properties as a list of `json` structures, each one having a **name**, **type**, **unit**, **description**, and if relevant, a list of **dimension**(s) (here called **dims**). Inserting the properties displayed in the table above, our Entity is complete and may look like
```

Comment:
Note:

Like for dimensions, SOFT7 suggests a new syntax expressing properties using a dict with the name as key:

```
properties: {
    "N": {
            "type":"str",
            "unit":"ns",
            "dims": ["N"],
            "description": "Time"
    },
    ...
}
```